COM-658 Program module - 3 new tabs for index testing

tmvumbi2 opened the pull request:
Jira ticket: https://jembiprojects.jira.com/browse/COM-658

lauravignoli committed:
COM-658 Make function more generic

Update getPatientMostRecentDefaulterProgramOutcome to getPatientMostRecentProgramOutcome.

lauravignoli committed:
COM-658 Add index testing queries and functions

tmvumbi2 committed:
COM-658 Add nb contacts and enrolled contacts to the index list
The commit also updates the index list to exclude index patients for which all the contacts have been enrolled into the index program

tmvumbi2 committed:
COM-658 Implement columns for index lists
Adds the implementation for all the missing columns on index testing lists

lauravignoli commented on configuration/globalproperties/indexTesting.xml:
```diff
@@ -0,0 +1,95 @@
+        getPatientMostRecentProgramAttributeValueFromName(p.person_id, 'PROGRAM_MANAGEMENT_2_NOTIFICATION_DATE') as 'PROGRAM_MODULE_LABEL_NOTIFICATION_DATE_KEY',
+        getPatientMostRecentProgramAttributeCodedValueFromName(p.person_id, 'PROGRAM_MANAGEMENT_3_NOTIFICATION_OUTCOME', ${language}) as 'PROGRAM_MODULE_LABEL_NOTIFICATION_OUTCOME_KEY',
+        getMostRecentDateObservation(p.person_id, 'HIV Test Date') as 'PROGRAM_MODULE_LABEL_TEST_DATE_KEY',
+        getMostRecentCodedObservation(p.person_id, 'Services', ${language}) as 'PROGRAM_MODULE_LABEL_TEST_PLACE_KEY',
```
@tmvumbi2 The correct Service/Location selected can be obtained from the obs location for the "Final Test Result" capture.

lauravignoli committed:
COM-658 Fixed GP definition

Used &lt; instead of <.

lauravignoli committed:
COM-658 Fix test place

The correct Service/Location for test place can be obtained from the obs location for the "Final Test Result" capture.

lauravignoli committed:
COM-658 Update index name and relationship columns

Show details of first index instead of concat all indexes.
